content/docs/command-reference/import-url.md

```diff
@@ -125,12 +125,15 @@ source.
   default file name: `<file>.dvc`, where `<file>` is the desired file name of
   the imported data (`out`).
 
-- `--no-exec` - create `.dvc` file without actually downloading `url`. E.g. if
-  the file or directory already exists, this can be used to skip the download.
-  The data hash is not calculated when this option is used, only the import
-  metadata is saved to the `.dvc` file. `dvc commit <out>.dvc` can be used if
-  the data hashes are needed in the `.dvc` file, and to save existing data to
-  the cache.
+- `--no-exec` - create the import `.dvc` file without actually downloading
+  `url`. No connection to the source location is made, so the data hash is not
+  calculated. Only the import operation is recorded in the `.dvc` file. Use
+  `dvc commit` to save any existing dep/out files to the cache and record their
+  hashes to the `.dvc` file.
+
+  This can be useful to skip the download if the file or directory already
```

Comment:
Seems it got too complicated ... I'm not sure I follow all the logic in these two paragraphs now

Reply:
I simplified it now as I just mentioned but it's still a little long. I don't see that as a big problem, but if you prefer I can remove the paragraph about motivation (added because you mentioned you liked the phrase "cook the imports before downloading altogether" or something like that).